src/fairseq2/checkpoint.py

```diff
@@ -32,14 +32,35 @@
 from torch.nn import Module
 
 from fairseq2.gang import Gang
+from fairseq2.logging import get_log_writer
 from fairseq2.models.utils.checkpoint import load_checkpoint
 from fairseq2.typing import CPU, DataClass, override
 from fairseq2.utils.dataclass import to_safe_dict
 
+log = get_log_writer(__name__)
```

Comment:
nit: maybe this should be called `logger` or `log_writer`. `log` is a confusing name for the writer